lib/rule-tester/flat-rule-tester.js

```diff
@@ -663,7 +675,31 @@ class FlatRuleTester {
                 }
             });
 
-            const schema = getRuleOptionsSchema(rule);
+            let schema;
+
+            try {
+                schema = getRuleOptionsSchema(rule);
+            } catch (err) {
+                err.message += metaSchemaDescription;
+                throw err;
+            }
+
+            /*
+             * Check and throw an error if the schema is an empty object (`schema:{}`), because such schema
+             * doesn't validate or enforce anything and is therefore considered a possible error. If the intent
+             * was to skip options validation, `schema:false` should be set instead (explicit opt-out).
+             *
+             * For this purpose, a schema object is considered empty if it doesn't have any own enumerable string-keyed
+             * properties. While `ajv.compile()` does use enumerable properties from the prototype chain as well,
+             * it caches compiled schemas by serializing only own enumerable properties, so it's generally not a good idea
+             * to use inherited properties in schemas because schemas that differ only in inherited properties would end up
+             * having the same cache entry that would be correct for only one of them.
+             *
+             * At this point, `schema` can only be an object or `null`.
+             */
+            if (schema && Object.keys(schema).length === 0) {
+                throw new Error(`\`schema: {}\` is a no-op${metaSchemaDescription}`);
```

Comment:
Disallowing `schema: {}` is most important since it's the most common, accidental, no-op schema.

I'm fine with this one check for now, but just want to ask if any other obvious/common no-op schemas come to mind? In the RFC, I mentioned `schema: { type: "array" }` as another no-op schema we could disallow, with the intention being to leave the door open to any others that we judge worth checking for. Obviously, we can only add update the banned no-op list during major versions.

Reply:
is there a related discussion in ajv repo? it could benefit all the dependents (not just eslint) - if it could be supported by ajv.

Reply:
I did a brief search and didn't immediately find a relevant discussion in in ajv. But I have included a lot more context in this issue discussing a potential lint rule:
* https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/296

Overall, we should probably just allow any schema in ESLint except extremely common and obvious mistakes like `{}`, and consider linting to detect further possible problems.

Reply:
Newer versions of Ajv have some built-in validations, but there were backwards compatibility issues when we tried to switch to Ajv 7+ (https://github.com/eslint/eslint/pull/13911). However, I believe `schema: { type: "array" }` would be considered valid by Ajv. It's just useless in ESLint because we're always passing arrays to these validations.

I agree that for now RuleTester should only check for the common and obvious mistake `schema: {}`.